hack/compile-bpf.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+UNAME=$(uname -r)
+ARCH=x86
+DIR=$(dirname $0)
+CFILE=${CFILE:-"${DIR}/../bpfassets/perf_event/perf_event.c"}
+OUTPUT=${OUTPUT:-"${DIR}/../_output/bpf-modules/perf_event.o"}
+VMLINUX_H=${VMLINUX_H:-"https://raw.githubusercontent.com/iovisor/bcc/master/libbpf-tools/x86/vmlinux_518.h"}
+if [ ! -f ${DIR}/../bpfassets/perf_event/perf_event.c ]; then
+    echo "perf_event.c not found"
+    exit 1
+fi
+if [ ! -f ${DIR}/../_output/bpf/vmlinux.h ]; then
+    echo "downloading vmlinux.h"
```

Comment:
This looks similar to what libebpf does.

btw, can we build this in one kernel/LinuxOS and run in a different kernel/LinuxOS?